### PR TITLE
Issue #30: Fixed bug "hand doesn't refill during momentum run"

### LINKED ISSUE #30: Hand doesn't refill during momentum run
Investigate bug where ending your turn during a momentum run doesn't refill your hand.

### PR DESCRIPTION
**Pull request by heckayla**
Closes #30. I fixed a bug that caused the hand to not refill with new cards properly when the turn is ended mid-momentum run, and ensured that (as far as I can tell) the hand gets refilled properly in all situations.

### CODE REVIEW

File: CardsForClimateUnityProj/Assets/Scripts/GameManager.cs [hunk 1]
```diff
@@ -394,9 +399,9 @@ public void EndTurn()
         //reset counter
         activePlayerCardCount = 0;
         //reset values
```
Comment (tannerwelsh) [inline CardsForClimateUnityProj/Assets/Scripts/GameManager.cs:401]: I wonder if we should encapsulate these value resets into a function, like `ResetValues()`?

File: CardsForClimateUnityProj/Assets/Scripts/GameManager.cs [hunk 2]
```diff
@@ -521,14 +527,14 @@ public bool PlayerCardsHope()
...
-
-        if((playedCard.hope > 0 || playedCard.hope < 0))
+        if (negativeHope >= 0 || hasMomentum) return true;
+        else if((playedCard.hope != 0))
         {
             hopeValid = true;
             return true;
         } else
         {
+            // If all of these attempts failed, then there is no hope here :(
```
Comment (tannerwelsh) [inline CardsForClimateUnityProj/Assets/Scripts/GameManager.cs:537]: 😭

File: CardsForClimateUnityProj/Assets/Scripts/GameManager.cs [hunk 3]
```diff
@@ -168,8 +170,10 @@ void Update()
         //Testing key for turn beginning
         if (!gameOver && Input.GetKeyDown(KeyCode.P))
         {
-            //Call begin turn to intialize event card
-            BeginTurn();
+            // Call end turn (if we're not starting the game for the first time) 
+            // and begin turn to intialize event card, redraw hand, and apply active cards
```
Comment (tannerwelsh) [inline CardsForClimateUnityProj/Assets/Scripts/GameManager.cs:174]: Thanks for these comments!

**Review by tannerwelsh — APPROVED** [review]
Changes look good to me, thanks for doing these!

File: CardsForClimateUnityProj/Assets/Scripts/GameManager.cs (hunk 1 above)
Comment (heckayla) [reply CardsForClimateUnityProj/Assets/Scripts/GameManager.cs:401]: That's a good point! Probably wouldn't hurt if one of us wants to do it alongside finishing an issue somewhere down the line

**Pull request merged**